core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -152,6 +154,7 @@ public static String toJson(TableMetadata metadata) {
     }
   }
 
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
I will do another PR to fix all the code complexity complaints in this class after this is merged.